build-system/common/git.js

```diff
@@ -27,6 +27,48 @@ const {
 } = require('./travis');
 const {getStdout} = require('./exec');
 
+/**
+ * Returns the upstream master branch, falling back to local master if upstream
+ * could not be determined.
+ * @return {string}
+ */
+function gitFindMaster() {
+  const upstreamRegex = /[:/]ampproject\/amphtml\.git$/i;
+
+  // Try obvious choices for upstream remote name first, hoping for a quick hit
+  // before iterating through remotes to find the correct one.
+
+  // Developer documentation recommends creation of 'upstream' remote
+  // https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#set-up-aliases-for-the-remote-git-repositories
+  const upstreamUrl = getStdout('git remote get-url upstream').trim();
+  if (upstreamRegex.test(upstreamUrl)) {
+    return 'upstream/master';
+  }
+  // Next, check the remote branch that local master is tracking
+  const trackingName = getStdout('git config branch.master.remote').trim();
+  if (trackingName) {
+    const trackingUrl = getStdout(`git remote get-url ${trackingName}`).trim();
+    if (upstreamRegex.test(trackingUrl)) {
+      return trackingName + '/master';
+    }
+  }
+  // Finally, iterate through remote names, checking URLs
+  const remoteNames = getStdout('git remote show')
+    .split(/\r?\n/)
+    .map((name) => name.trim())
+    .filter(Boolean);
+  while (remoteNames.length) {
+    const remoteName = remoteNames.pop();
+    const remoteUrl = getStdout(`git remote get-url ${remoteName}`).trim();
+    if (upstreamRegex.test(remoteUrl)) {
+      return remoteName + '/master';
+    }
+  }
```

Comment:
Style nit: Can we use `for (const remoteName in remoteNames) {...}` instead of the `while` loop?